lib/LedgerNodePeers.js

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Ledger node operations management class.
+ *
+ * Copyright (c) 2017-2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const bedrock = require('bedrock');
+const database = require('bedrock-mongodb');
+const {util: {BedrockError}} = bedrock;
+const {COLLLECTION_NAME} = require('./ledgerNodePeerCollection');
+
+/**
+ * The LedgerNodePeers class exposes peer information.
+ */
+module.exports = class LedgerNodePeers {
+  constructor({ledgerNode}) {
+    this.ledgerNode = ledgerNode;
+    this.ledgerNodeId = ledgerNode.id;
+  }
+
+  async add({peer}) {
+    const collection = database.collections[COLLLECTION_NAME];
+    const now = Date.now();
+    const peerRecord = {
+      meta: {
+        created: now,
+        ledgerNodeId: this.ledgerNodeId,
+        updated: now
+      },
+      peer,
+    };
+    let result;
+    try {
+      result = await collection.insertOne(peerRecord);
+    } catch(e) {
+      if(database.isDuplicateError(e)) {
+        throw new BedrockError(
+          'Duplicate peer.',
+          'DuplicateError', {
+            httpStatusCode: 409,
+            public: true,
+            ledgerNodeId: this.ledgerNodeId,
+            peerId: peer.id,
+          }, e);
+      }
+    }
+
+    return result.ops[0];
```

Comment:
this can fail if the error is not a `isDuplicateError` then no error is thrown and result will be `undefined`

Reply:
Fixed.